### PR TITLE
#34 crud-operations-for-admin 

### PR DESCRIPTION
**Pull request by artempenteskul**

### CODE REVIEW

**Comment by SerhiiPenteliuk** [comment]
Add new lines where necessary

**Review by SerhiiPenteliuk — CHANGES_REQUESTED** [review]
add new lines where necessary

File: softserve/app.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+from flask import Flask, render_template, request, redirect
+from flask_pymongo import PyMongo
+from bson.objectid import ObjectId
```
Comment (ivan-dubei) [inline softserve/app.py:3]: list imports in sorted order

File: softserve/app.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+app = Flask(__name__)
+app.config['MONGO_URI'] = "mongodb://localhost:27017/softserve"
```
Comment (ivan-dubei) [inline softserve/app.py:6]: move this value into config file

File: softserve/app.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+@app.route('/books-storage/<string:_id>')
+def books_storage_details(_id):
+    book = books_collection.find_one({'_id': ObjectId(_id)})
+    return render_template('books-storage-details.html', book=book)
+
+
+@app.route('/books-storage/<string:_id>/del')
+def books_storage_del(_id):
+    book = books_collection.find_one({'_id': ObjectId(_id)})
+    try:
```
Comment (ivan-dubei) [inline softserve/app.py:32]: Do we need a bare try-exept here ? May be we may add some check for 'book' ?

File: softserve/app.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        number_of_pages = request.form['number_of_pages']
+        description = request.form['description']
+        language = request.form['language']
+
+        try:
+            books_collection.update(book, {'title': title, 'author': author, 'genre': genre, 'year': year,
+                                           'language': language, 'publisher': publisher,
+                                           'number_of_pages': number_of_pages, 'description': description})
+            return redirect('/books-storage')
+        except:
```
Comment (ivan-dubei) [inline softserve/app.py:57]: We may add some better exception handling.
Something like:
    except Exception as e:
        <add some message with error info here>

File: softserve/app.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        number_of_pages = request.form['number_of_pages']
+        description = request.form['description']
+        language = request.form['language']
+
+        try:
+            books_collection.insert_one({'title': title, 'author': author, 'genre': genre, 'year': year,
+                                           'language': language, 'publisher': publisher,
+                                           'number_of_pages': number_of_pages, 'description': description})
+            return redirect('/books-storage')
+        except:
```
Comment (ivan-dubei) [inline softserve/app.py:81]: Add better errors handling

File: softserve/app.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        author = request.form['author']
+        genre = request.form['genre']
+        year = request.form['year']
+        publisher = request.form['publisher']
+        number_of_pages = request.form['number_of_pages']
+        description = request.form['description']
+        language = request.form['language']
+
+        try:
+            books_collection.update(book, {'title': title, 'author': author, 'genre': genre, 'year': year,
```
Comment (Palusha) [inline softserve/app.py:53]: Use update_one() instead of update()

File: softserve/app.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+def books_storage_details(_id):
+    book = books_collection.find_one({'_id': ObjectId(_id)})
+    return render_template('books-storage-details.html', book=book)
+
+
+@app.route('/books-storage/<string:_id>/del')
+def books_storage_del(_id):
+    book = books_collection.find_one({'_id': ObjectId(_id)})
+    try:
+        books_collection.delete_one(book)
```
Comment (Palusha) [inline softserve/app.py:33]: Better to use update_one() for book status instead of delete_one()

**Review by SerhiiPenteliuk — APPROVED** [review]

**Comment by ivan-dubei** [comment]
Change application name to another one, e.g to the `book_storage`, like it is done in the neighboring pool request.

**Comment by mehalyna** [comment]
non actual PR

**Pull request closed without merging**